rfcs/0000-rlp-v2.md

```diff
@@ -0,0 +1,1028 @@
+kind: RateLimitPolicy
+metadata:
+  name: toystore-non-admin-users
+spec:
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: toystore
+  limits:
+    - name: readers
+      matches:
+        - path:
+            type: PathPrefix
+            value: "/toys"
+          method: GET
+      rates:
+        - limit: 50
+          duration: 1
+          unit: minute
+      when:
+        - selector: auth.identity.group
+          operator: neq
+          value: admin
+
+    - name: writers
+      matches:
+        - path:
+            type: PathPrefix
+            value: "/toys"
+          method: POST
+      rates:
+        - limit: 5
+          duration: 1
+          unit: minute
+      when:
+        - selector: auth.identity.group
+          operator: neq
+          value: admin
+```
+
+Apart from the descriptor action (associated with both routes):
+
+```yaml
+- metadata:
+    descriptor_key: "auth.identity.group"
+    metadata_key:
+      key: "envoy.filters.http.ext_authz"
+      path:
+        - segment:
+            key: "identity"
+        - segment:
+            key: "group"
+```
+
+...and its corresponding Limitador condition:
+
+```
+auth.identity.group != "admin"
+```
+
+The following additional artificial descriptor actions will be generated:
+
+```yaml
+# associated with route GET /toys*
+- generic_key:
+    descriptor_key: "context.request.http.route_matcher"
+    descriptor_value: "2718701ec9bfd79132e58d92aed722489443094bf9c616e1b74361fe68360f05" # SHA256 hashing of [{"path":{"type":"PathPrefix","value":"/toys"},"method":"GET"}]
+
+# associated with route POST /toys*
+- generic_key:
+    descriptor_key: "context.request.http.route_matcher"
+    descriptor_value: "4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d" # SHA256 hashing of [{"path":{"type":"PathPrefix","value":"/toys"},"method":"POST"}]
+```
+
+...and their corresponding Limitador conditions.
+
+In the end, the following Limitador configuration is yielded:
+
+```yaml
+- conditions:
+    - context.request.http.route_matcher == "2718701ec9bfd79132e58d92aed722489443094bf9c616e1b74361fe68360f05"
+    - auth.identity.group != "admin"
+  max_value: 50
+  seconds: 60
+  namespace: "*.toystore.com"
+
+- conditions:
+    - context.request.http.route_matcher == "4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d"
+    - auth.identity.group != "admin"
+  max_value: 5
+  seconds: 60
+  namespace: "*.toystore.com"
+```
+
+To improve readability, the unique hash identifiers can be suffixed with a plain identifier of the RLP itself (namespace and name) and the limit (limit name when available, limit index otherwise). E.g.: the two unique identifiers from above could become respectively `2718701ec9bfd79132e58d92aed722489443094bf9c616e1b74361fe68360f05--toystore/toystore-non-admin-users/readers` and `4f70fc57ad52a2664e3920f373633a9b2b2f4f58f17b39a8d3a3a3485fd91c4d--toystore/toystore-non-admin-users/writers`.
```

Comment:
This is more than about readability, `{"path":{"type":"PathPrefix","value":"/toys"},"method":"POST"}` could exist on multiple `HTTPRoute`s or `HTTPRouteRule`s we need to differentiate between them. 

Maybe flip the ordering? `<user-parseable>--<sha>` to improve readability even more?

Reply:
Thanks! I've made sure to reflect this in the text now.